src/Perl6/Actions.nqp

```diff
@@ -6613,35 +6615,26 @@ class Perl6::Actions is HLL::Actions does STDActions {
 
     method term:sym<value>($/) { make $<value>.ast; }
 
-    sub handle-list-semis($/, $past) {
+    method circumfix:sym<( )>($/) {
+        my $Pair := $*W.find_symbol(['Pair']);
+        my $past := $<semilist>.ast;
+
         if !+$past.list {
             $past := QAST::Stmts.new( :node($/) );
             $past.push(QAST::Op.new( :op('call'), :name('&infix:<,>'), :node($/)));
         }
         # Look for any chained adverb pairs and relocate them.
         # Try to reuse existing QAST where possible.
         elsif $*FAKE_INFIX_FOUND {
-            my @EXPR;
-            my $semis := $<semilist><statement>;
-            my $numsemis := +$semis;
-
-            my $i := -1;
-            while ++$i < $numsemis {
-                my $EXPR := $semis[$i]<EXPR>;
-                if nqp::defined($EXPR) {
-                    @EXPR.push($EXPR);
-                }
-            }
-            $numsemis := +@EXPR;
-
+            my $numsemis := +$<semilist><statement>;
             if $numsemis > 1 {
                 $past := QAST::Stmts.new( :node($/) );
                 $past.push(QAST::Op.new( :op('call'), :name('&infix:<,>'), :node($/)));
             }
-
-            my $semi := -1;
-            while ++$semi < $numsemis {
-                my $EXPR := @EXPR[$semi];
+            my $semi := 0;
```

Comment:
Feels like you missed a commit (a26e95bab41e7b787c) that changed this.  Is there a reason you want to go back to the old behaviour here?

Reply:
That's confusing. The commit is actually there vrurg/rakudo@a26e95bab41e7b787c8283f7ae24ae3859727cfe. Perhaps a result of chained migration macbook -> mac mini -> macbook with help of backups and rsync.

Reply:
I have re-applied related commits by cherry-picking them.